apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -162,44 +211,17 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
                     </VerticalList>
                 </div>
             </div>
-            <div className="grow overflow-auto border-gray-45 pt-5 md:w-2/5 md:border-l md:pl-7">
-                <TabGroup size="md">
-                    <TabList>
-                        <Tab>Bytecode</Tab>
-                    </TabList>
-                    <TabPanels>
-                        <TabPanel>
-                            <div className="h-verticalListLong overflow-auto">
-                                <ModuleViewWrapper
-                                    id={id}
-                                    modules={modules}
-                                    selectedModuleName={selectedModule}
-                                />
-                            </div>
-                        </TabPanel>
-                    </TabPanels>
-                </TabGroup>
+            <div className="hidden w-4/5 md:block">
+                <SplitPanes
+                    direction="horizontal"
+                    defaultSizes={[40, 60]}
+                    panels={bytecodeContent}
+                />
             </div>
-            <div className="grow overflow-auto border-gray-45 pt-5 md:w-3/5 md:border-l md:pl-7">
-                <TabGroup size="md">
-                    <TabList>
-                        <Tab>Execute</Tab>
-                    </TabList>
-                    <TabPanels>
-                        <TabPanel>
-                            <div className="h-verticalListLong overflow-auto">
-                                {id && selectedModule ? (
-                                    <ModuleFunctionsInteraction
-                                        // force recreating everything when we change modules
-                                        key={`${id}-${selectedModule}`}
-                                        packageId={id}
-                                        moduleName={selectedModule}
-                                    />
-                                ) : null}
-                            </div>
-                        </TabPanel>
-                    </TabPanels>
-                </TabGroup>
+            <div className="md:hidden">
```

Comment:
So generally, I think this is the right way to tackle this stuff responsively. However, I think keeping multiple copies of the syntax highligher and execution panel in the DOM tree might be annoyingly expensive. Given these depend on client-side features, I think it'd be okay if we just did this with a media query + state (probably a `useMediaQuery` hook or something), to just conditionally render (either setting orientation to vertical, or just disabling the split)

Reply:
@Jordan-Mysten could you take a look again when you get a chance, I added 2 new hooks to resolve the screensizing check. This is similar to the implementation used inside `tanstack query` but unfortunately we cannot export that method out.